networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/discovery/discv5/NodeRecordConverter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.p2p.discovery.discv5;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.ethereum.beacon.discovery.schema.EnrField;
+import org.ethereum.beacon.discovery.schema.NodeRecord;
+import tech.pegasys.artemis.networking.p2p.discovery.DiscoveryNodeId;
+import tech.pegasys.artemis.networking.p2p.discovery.DiscoveryPeer;
+
+public class NodeRecordConverter {
+  private static final Logger LOG = LogManager.getLogger();
+
+  static Optional<DiscoveryPeer> convertToDiscoveryPeer(final NodeRecord nodeRecord) {
+    return addressFromFields(nodeRecord, EnrField.IP_V4, EnrField.TCP_V4)
+        .or(() -> addressFromFields(nodeRecord, EnrField.IP_V6, EnrField.TCP_V6))
+        .or(() -> addressFromFields(nodeRecord, EnrField.IP_V4, EnrField.TCP_V6))
```

Comment:
I think this should be:
```suggestion
        .or(() -> addressFromFields(nodeRecord, EnrField.IP_V6, EnrField.TCP_V4))
```

Reply:
We should probably rename `TCP_V4` -> `TCP` and `UDP_V4` -> `UDP`

Reply:
Yeah I misread your comment and once I had it in my head the wrong way no amount of spec reading could fix it...  Rename will have to happen in discovery and flow through.  Will try to get that into the next release.